Zend/zend_string.h

```diff
@@ -241,7 +241,9 @@ static zend_always_inline zend_string *zend_string_dup(zend_string *s, bool pers
 	if (ZSTR_IS_INTERNED(s)) {
 		return s;
 	} else {
-		return zend_string_init(ZSTR_VAL(s), ZSTR_LEN(s), persistent);
+		zend_string *res = zend_string_init(ZSTR_VAL(s), ZSTR_LEN(s), persistent);
```

Comment:
this is not needed for `implode` - is there any way to test it easily from php?

Reply:
Could you please clarify what you mean?
If you want to read some internal data from PHP from within a test, you can always use the zend-test extension to do so.